OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -285,13 +285,17 @@ public void onException(Exception ex) {
                             if (key.equalsIgnoreCase(fieldView.getField().getName())) {
                                 List<RegisterResponseFieldError> error = messageBody.get(key);
                                 showErrorOnField(error, fieldView);
-                                fieldErrorShown = true;
+                                if (!fieldErrorShown) {
```

Comment:
Based on this `bool` we are only showing the popup, so, my suggested name `errorDialogShown` is a better option.